nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -457,6 +486,19 @@ in {
                 return trim(file_get_contents($file));
               }
             ''}
+            ${optionalString (cfg.secretFile != null) ''
+              function nix_read_secrets() {
```

Comment:
This should be deduplicated IMHO. A `nix_read_file($filename)` could be made for instance.

Reply:
done (with an additional parameter $error to model error messages in case the file could not be read)

Reply:
note that there's now two functions: the nix_read_file function suggested here, and a nix_decode_json_file function which does the same but also handles json decoding & potential errors